source/slang/slang-check-decl.cpp

```diff
@@ -1053,6 +1053,33 @@ namespace Slang
                 addModifier(varDecl, m_astBuilder->create<NVAPIMagicModifier>());
             }
         }
+
+        if (auto interfaceDecl = as<InterfaceDecl>(varDecl->parentDecl))
+        {
+            if (auto basicType = as<BasicExpressionType>(varDecl->getType()))
+            {
+                switch (basicType->baseType)
+                {
```

Comment:
Could consider using `const BaseTypeInfo& info = BaseTypeInfo::getInfo(baseType);` will indicate intness, and will be kept hopefully in sync if any new int types are added.